### ISSUE TITLE
Migrate demos to new API

### ISSUE THREAD

**Issue by tafsiri** [tensorflow/tfjs-core#651]
TODO: This issue will need a checklist to track all the demos that we want to convert.

**Comment by nsthorat** [comment]
This doesn't have to happen before 0.5.0 since we're going to support backwards compatibility for this version. Specifically, we're missing a few optimizers for the model builder to be fully converted. We can tackle this after 0.5.0.

**Comment by tafsiri** [comment]
No longer needed, as we now have tfjs-examples
